Code/Framework/AzToolsFramework/Tests/Prefab/PrefabUpdateInstancesTests.cpp

```diff
@@ -87,9 +87,9 @@ namespace UnitTest
 
         // Add another entity to the Instance and use it to update the PrefabDom of Template.
         AZ::Entity* entity2 = CreateEntity("Entity 2");
-        newInstance->AddEntity(*entity2);
         AzToolsFramework::EditorEntityContextRequestBus::Broadcast(
             &AzToolsFramework::EditorEntityContextRequests::HandleEntitiesAdded, AzToolsFramework::EntityList{entity2});
```

Comment:
I notice that if `HandleEntitiesAdded` gets called after AddEntity. During scrubbing, the entity will be marked as dirty, then GenerateUndoNodesForEntityChangeAndUpdateCache will be invoked for the new entity, and eventually generator  complains that it cannot find the entity DOM in template.

Like other places in this test file, I believe it should be called immediately after entity creation. Looking into the HandleEntitiesAdded code, it corresponds with entity creation, not something else.